### ISSUE TITLE
popup preview area is freeze unexpectedly

### ISSUE THREAD

**Issue by the-liquid-metal** [bartekmotyl/simple-video-cutter#47]
after several click on timeline and toolbar, popup preview area is freeze unexpectedly.

**Comment by bartekmotyl** [comment]
This a know issue. I mean - I have seen this few times, but I have no reliable replication procedure in order to debug it. 
If you can provide one it would be great.

**Comment by bartekmotyl** [comment]
I have done some improvements in version 0.20 and the problem should be gone...

**Comment by bartekmotyl** [comment]
fixed in development branch, will be included in next release

**Comment by 0lm** [comment]
this bug seems to be back. how to reproduce: move mouse cursor fast over the timeline. eventually it will quickly freeze on a random frame.

when i moved mouse slow and very carefully, it doesnt seem to freeze. but when you do it a bit faster (and this happens often, since you want to check frames quickly for the right timestamp) then it eventually freezes when you move mouse too fast around the timeline.

**Comment by bartekmotyl** [comment]
Thanks for reporting, I was also able to reproduce the issue (not easily though).

**Comment by bartekmotyl** [comment]
I have just committed another approach to fix this problem - debounce updates of the preview (with 100ms delay). Looks promising on my machine. I will include the fix in next release.

**Comment by 0lm** [comment]
> Thanks for reporting, I was also able to reproduce the issue (not easily though).

After thinking a little bit about what you replied here and testing things myself, I understand now why you said "(not easily though)".

I tested (on v0.24) different videos of mine, including the one where it happened very easily and alot. Conclusion: other videos I tested it was really hard to reproduce this same bug. Not sure why. But evetually after a longer while, other videos also finally froze. Then I tested the other video again, where it happened alot, and it could be reproduced very easily on it, it froze imediately when moving mouse a bit too fast.

I guess it has to do with some certain specs videos can have, so it was harder for you to reproduce it. 

But testing again with v0.25, the results were: it didnt even freeze anymore with the one video, that was really easy to make freeze, thank you!
